azurerm/internal/services/compute/resource_arm_dedicated_host_group.go

```diff
@@ -0,0 +1,203 @@
+package compute
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDedicatedHostGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDedicatedHostGroupCreate,
+		Read:   resourceArmDedicatedHostGroupRead,
+		Update: resourceArmDedicatedHostGroupUpdate,
+		Delete: resourceArmDedicatedHostGroupDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[^_\W][\w-.]{0,78}[\w]$`), ""),
+			},
+
+			"location": azure.SchemaLocation(),
+
+			// There's a bug in the Azure API where this is returned in upper-case
+			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/8068
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"platform_fault_domain_count": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(1, 3),
+			},
+
+			"zones": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[1-9]\d*$`), ""),
+				},
+				// Currently only one endpoint is allowed.
+				// we'll leave this open to enhancement when they add multiple zones support.
+				MaxItems: 1,
+			},
```

Comment:
we can use `SingleZoneSchema` here instead

Reply:
What if azure supports multiple zones later? Then should we introduce a breaking change?

Reply:
No, it would still be the same name and type, just a different min/max item.

Reply:
Ooh, I mis-understood the comment 😨